todos/src/components/Todo.jsx

```diff
@@ -0,0 +1,27 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTrash } from "@fortawesome/free-solid-svg-icons";
+import { useDispatch } from "react-redux";
+import { endTask, deleteTask } from "../redux/todoSlice";
+function Todo(props) {
+  const dispatch = useDispatch();
+
+  return (
+    <div className="w-1/2 flex items-center justify-between py-2 my-4 border-b-2 border-gray-200 m-auto">
+      <div className="flex space-x-2 items-center">
+        <input
+          onChange={() => dispatch(endTask(props.index))}
+          type="checkbox"
+        />
```

Comment:
You don't even need to use the index, you can just use the todo `id` and in your `endTask` reducer you will find the todo with that `id` and end it

Reply:
Did you check this